### PR TITLE
Bug fix in LB rules cleanup when selector rule is present

### PR DESCRIPTION
**Pull request by alena1108**
https://github.com/rancher/rancher/issues/9118

### CODE REVIEW

File: code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java [hunk 1]
```diff
@@ -469,7 +469,7 @@ protected void updateObjectEndPoints(final Object object, final String resourceT
                     .toString());
         }
 
-        if (oldPortToIp.contains(newPortToIp) && newPortToIp.contains(oldPortToIp)) {
+        if (oldPortToIp.containsAll(newPortToIp) && newPortToIp.containsAll(oldPortToIp)) {
```
Comment (alena1108) [inline code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/service/impl/ServiceDiscoveryServiceImpl.java]: this line is unrelated to the bug fix, just addressed the warning

**Comment by cjellick** [comment]
LGTM
